ship_functions.py

```diff
@@ -0,0 +1,212 @@
+                                                      stratify=y)
+    images_transfert(X_train, o_folder, p_folder + "train/")
+    images_transfert(X_val, o_folder, p_folder + "val/")
+
+    return X_train, X_val
+
+
+def weights_selection(proj_dir, choice):
+    '''Manage the weights file with returning the path after downloading file if
+    necessary'''
+        # Select weights file to load
+    if choice.lower() == "coco":
+        weights_path = proj_dir + "coco_trained_weights.h5"
+        # Download weights file
+        if not os.path.exists(weights_path):
+            utils.download_trained_weights(weights_path)
+    elif choice.lower() == "last":
+        # Find last trained weights
+        weights_path = model.find_last()
+    elif choice.lower() == "imagenet":
+        # Start from ImageNet trained weights
+        weights_path = model.get_imagenet_weights()
+    else:
+        weights_path = choice
+
+    return weights_path
+
+
+def train_configuration(logs_path, weights_path, weights):
+    '''Configure the MRCNN algorithme for the training'''
+    config = ship_config.ShipConfig()
+    config.display()
+
+    ship_model = model.MaskRCNN(mode="training", config=config,
+                                model_dir=logs_path)
+
+    #Exclude the last COCO layers because they require a matching number of classes
+    if weights.lower() == "coco":
+        ship_model.load_weights(weights_path, by_name=True, exclude=[
+            "mrcnn_class_logits", "mrcnn_bbox_fc",
+            "mrcnn_bbox", "mrcnn_mask"])
+    else:
+        ship_model.load_weights(weights_path, by_name=True)
+
+    return ship_model, config
+
+
+def launch_training(dataset_dir, train, val, local_model, config):
+    '''Launch Mask RCNN training'''
+    # Training dataset.
+    dataset_train = ship_dataset.ShipDataset()
+    dataset_train.load_ship(dataset_dir + "train/", train)
+    dataset_train.prepare()
+
+    # Validation dataset
+    dataset_val = ship_dataset.ShipDataset()
+    dataset_val.load_ship(dataset_dir + "val/", val)
+    dataset_val.prepare()
+
+    # Beginning of the training
+    print("Training network heads")
+    local_model.train(dataset_train, dataset_val,
+                      learning_rate=config.LEARNING_RATE,
+                      epochs=110,
+                      layers='heads')
+
+
+def detect_configuration(result_folder, weights_path):
+    '''Configure the MRCNN algorithme for the training'''
+    config = ship_config.ShipConfig()
+    config.display()
+
+    ship_model = model.MaskRCNN(mode="inference", config=config,
+                                model_dir=result_folder)
+
+    ship_model.load_weights(weights_path, by_name=True)
+
+    return ship_model, config
+
+
+def mask_analyse(mask):
+    '''Analyse the mrcnn mask result and return a string if exist'''
+    # We're treating all instances as one, so collapse the mask into one layer
+    mask = (np.sum(mask, -1, keepdims=True) >= 1)
+
+    if True in mask:
```

Comment:
That's a highly non-conventional use of `numpy`. I would use `mask.any()`. (Unless the numpy documentation says that the ` in ` syntax on an `ndarray` is good)